### PR TITLE
Initial work on handling FHIR requests in OCLAPI

### PR DESCRIPTION
**Pull request by davetrig**

### CODE REVIEW

File: ocl/fhir/mixins.py [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+class BaseFhirMixin(object):
+    def get_from_api(self, url):
+
+        #profile = UserProfile.objects.get(mnemonic=username)
+        #api_url = 'http://api.openconceptlab.org:8000'
+        #api_token = profile.user.auth_token.key
+
+        #api_url = 'http://api.openconceptlab.org:8000'
+        api_url = 'http://localhost:8000'
+        api_token = '891b4b17feab99f3ff7e5b5d04ccc5da7aa96da6'
```
Comment (paynejd) [inline ocl/fhir/mixins.py]: This should get the API token from the user's request, if present. If there is no API token in the request, then no API token should be used (which means the request may return unauthorized)

File: ocl/fhir/mixins.py [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+                group.element = group_elements
+                cm.group = [group]
+
+        return cm
+
+
+def get_equivalence(map_type):
+
+    if (map_type == 'Same As'):
```
Comment (paynejd) [inline ocl/fhir/mixins.py]: make this case insensitive

File: ocl/fhir/mixins.py (hunk 1 above)
Comment (rkorytkowski) [reply ocl/fhir/mixins.py]: Please see e.g. https://github.com/OpenConceptLab/oclapi/blob/master/ocl/manage/views.py#L95

File: ocl/fhir/mixins.py [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+class BaseFhirMixin(object):
+    def get_from_api(self, url):
+
+        #profile = UserProfile.objects.get(mnemonic=username)
+        #api_url = 'http://api.openconceptlab.org:8000'
+        #api_token = profile.user.auth_token.key
+
+        #api_url = 'http://api.openconceptlab.org:8000'
+        api_url = 'http://localhost:8000'
```
Comment (rkorytkowski) [inline ocl/fhir/mixins.py]: This is actually correct. I forgot that I used api.open.... for async tasks, which are run in celery (different service).

**Comment by rkorytkowski** [comment]
Once privileges issue is fixed we can merge that and deploy to the demo env. @davetrig do you think you could fix that today?

**Comment by davetrig** [comment]
> Once privileges issue is fixed we can merge that and deploy to the demo env. @davetrig do you think you could fix that today?

I should be able to make those changes this evening and resubmit.

**Comment by davetrig** [comment]
@rkorytkowski , @paynejd : I have committed the requested changes. Should be ready to go now.

**Comment by Karuhanga** [comment]
@davetrig @rkorytkowski it seems like the `fhir` name interferes with the `fhir` module in the dependencies. I am considering renaming it.

**Pull request merged**